lib/stripe/subscriptions/invoice.ex

```diff
@@ -88,9 +88,9 @@ defmodule Stripe.Invoice do
           })
 
   @type invoice_settings :: %{
-          default_payment_method: String.t() | nil,
-          custom_fields: custom_fields | nil,
-          footer: String.t() | nil
+          optional(:default_payment_method) => String.t() | nil,
+          optional(:custom_fields) => custom_fields | nil,
+          optional(:footer) => String.t() | nil
```

Comment:
👋 @steffenix THanks for the PR!  In `Customer.update/3`, we have `invoice_settings` listed as optional.  Are you seeing an explicit error somewhere?

Reply:
Using dializee with only one of the three parameters will tell me this function will fail because it's missing two parameters

Reply:
I forgot about this! `optional(key_type) => value_type`.  We can remove `| nil` from these and this PR is g2g!